src/main/java/com/sucy/skill/listener/ItemListener.java

```diff
@@ -137,24 +155,55 @@ public void onWorld(PlayerChangedWorldEvent event) {
     }
 
     @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
-    public void onHeld(PlayerItemHeldEvent event) {
-        if (SkillAPI.getSettings().isWorldEnabled(event.getPlayer().getWorld()))
-            SkillAPI.schedule(new UpdateTask(event.getPlayer(), true), 1);
+    public void onHeld(PlayerItemHeldEvent event)
+    {
+        Player player = event.getPlayer();
+        if (SkillAPI.getSettings().isWorldEnabled(player.getWorld())) {
+            PlayerData playerData = SkillAPI.getPlayerData(player);
+            if(playerData.getEquips().updateHandHeldItem(player, player.getInventory().getItem(event.getNewSlot()))) {
+                playerData.updatePlayerStat(player);
+            }
+        }
     }
 
     @EventHandler
-    public void onClose(InventoryCloseEvent event) {
-        if (SkillAPI.getSettings().isWorldEnabled(event.getPlayer().getWorld()))
-            SkillAPI.schedule(new UpdateTask((Player) event.getPlayer(), false), 1);
+    public void onClose(InventoryCloseEvent event)
+    {
+        Player player = (Player) event.getPlayer();
+        if (SkillAPI.getSettings().isWorldEnabled(player.getWorld())) {
+            PlayerData playerData = SkillAPI.getPlayerData(player);
+            if(playerData.getEquips().update(player)) {
+                playerData.updatePlayerStat(player);
+            }
+        }
     }
 
-    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
-    public void onInteract(PlayerInteractEvent event) {
-        if (SkillAPI.getSettings().isWorldEnabled(event.getPlayer().getWorld())
+    @EventHandler
+    public void onArmorEquip(PlayerInteractEvent event)
+    {
+        Player player = event.getPlayer();
+        ItemStack item = event.getItem();
+
+        if (SkillAPI.getSettings().isWorldEnabled(player.getWorld())
                 && (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK)
-                && event.getPlayer().getItemInHand() != null
-                && ARMOR_TYPES.contains(event.getPlayer().getItemInHand().getType())) {
-            SkillAPI.schedule(new UpdateTask(event.getPlayer(), false), 1);
+                && item != null
+                && ARMOR_TYPES.contains(item.getType()))
+        {
+            int equippedSlot = -1;
+
+            if(item.getType().name().contains("HELMET")) {
+                equippedSlot = 39;
+            }else if(item.getType().name().contains("CHESTPLATE")) {
+                equippedSlot = 38;
+            }else if(item.getType().name().contains("LEGGINGS")) {
+                equippedSlot = 38;
```

Comment:
Leggings should be 37 and Boots should be 36, no?

Reply:
opps, probably drunk that time lol, fixed